src/app/pages/history/table/table.component.ts

```diff
@@ -26,6 +26,11 @@ export class TableComponent {
     this.dataSource.paginator = this.paginator;
   }
 
+  public ngOnChanges(): void {
+    this.dataSource.sort = this.sort;
+    this.dataSource.paginator = this.paginator;
+  }
+
```

Comment:
Data table with sorting, pagination, and filtering - example 
not correct Lifecycle hook
https://material.angular.io/components/table/examples

Reply:
I used ngAfterViewInit, like in the example, but I add second lifecycle hook for detecting changes after we add a new event in the array

Reply:
what you must do after you change data?